src/azul/lambdas.py

```diff
@@ -98,12 +98,27 @@ def _lambda(self):
         return aws.lambda_
 
     def list_lambdas(self) -> list[Lambda]:
+        """
+        Return a list of all the Lambda functions in the AWS project
+        """
         return [
             Lambda.from_response(function)
             for response in self._lambda.get_paginator('list_functions').paginate()
             for function in response['Functions']
         ]
 
+    def list_deployment_lambdas(self) -> list[Lambda]:
```

Comment:
Good catch that the script should only iterate over lambdas for the current deployment. However I think we should avoid duplicating the implementation of `list_lambdas` here. I can think of two ways to do this:

1. Combine them into a single method with an optional parameter to control the filtering. This could either be a boolean or the name of the deployment, e.g. `def list_lambdas(self, deployment_only: bool = False) -> ...` or `def list_lambdas(self, deployment_name: Optional[str] = None) -> ...`.
2. Add a `deployment: Optional[str]` attribute to the `Lambda` class which is populated analogously to the existing `slot_location` attribute. The `list_deployment_lambdas` method can then call `list_lambdas` and filter the result based on that attribute.

Regardless, there should some sort of insurance in place so that the script doesn't crash if we encounter a Lambda where this variable isn't set.